apis/core/v1alpha2/readinessprovider_webhook.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha2
+
+import (
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var readinessproviderlog = logf.Log.WithName("readinessprovider-resource").WithValues("apigroup", "core")
+
+// SetupWebhookWithManager adds the webhook to the manager.
+func (r *ReadinessProvider) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	s, err := getScheme()
+	if err != nil {
+		return err
+	}
+
+	cl, err = client.New(mgr.GetConfig(), client.Options{Scheme: s})
+	if err != nil {
+		return err
+	}
+
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+var _ webhook.Validator = &ReadinessProvider{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *ReadinessProvider) ValidateCreate() error {
+	readinessproviderlog.Info("validate create", "name", r.Name)
+
+	for _, condition := range r.Spec.Conditions {
+		if condition.ResourceExistenceCondition == nil {
+			return apierrors.NewBadRequest(fmt.Sprintf("Expected condition %s to have exactly one type defined", condition.Name))
+		}
+	}
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *ReadinessProvider) ValidateUpdate(old runtime.Object) error {
+	readinessproviderlog.Info("validate update", "name", r.Name)
+
+	oldObj, ok := old.(*ReadinessProvider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected FeatureGate object, but got object of type %T", old))
+	}
+	if oldObj == nil {
+		return nil
+	}
+
```

Comment:
We still need to validate here all the fields we validate in create as well.

Reply:
Done